### ISSUE TITLE
Proposal for Governance

### ISSUE THREAD

**Issue by ashepherd** [ESIPFed/science-on-schema.org#7]
In review of [Schema.org - How We Work](https://schema.org/docs/howwework.html), here are my thoughts on how we could govern improvements to this repo.

1. Establish a Steering Group that has the ability to publish release candidates, hotfixes to production, and publish release candidates as new versions of production.
2. Use the Github Issue queue to accept proposals on improvements to guidelines and extension terms and models. 
3. Use the ESIP [Semantic Technologies Committee telecon](http://wiki.esipfed.org/index.php/Semantic_Technologies) to highlight new proposals allowing folks to engage with the Github issues they are interested in without having to follow all issues on the repo.
4. For documenting strategies and important decisions we adopt the [Architecture Decision Record framework](/ESIPFed/science-on-schema.org/tree/master/decisions) for summing up Github Issues that are lengthy and do not communicate decision points.

## 1. Where are improvements discussed? ##

* The [Github issue queue](https://github.com/ESIPFed/science-on-schema.org/issues) is where improvements to the vocabulary and documentation are initiated
* All new issues and group decisions could happen as part of the monthly [ESIP Semantic Technologies Committee telecon](GOVERNANCE.md#esip-swc-telecon)

## 2. Steering Group ##
schema.org has a steering group that collectively decides on what makes it in or not. They meet periodically to graduate issues into release candidates and from release candidates to production. 

__Q: Should we adopt a similar workflow? If so, how many people should sit on the committee?__

### Production ###

The master branch of this repository will be the production version of our guidelines and extensions. Pending changes that roll into production constitutes a new release with a versioned tag.

A rolling CHANGELOG.md file should be kept up to date as part of each new release.

### Release Candidate ###

Work on the documentation and extensions can happen in a development branch. At certain intervals, the development branch is pushed to a new release candidate branch which then migrates into a release candidate branch. Github Issue queue can reference the release candidate branch for comment from the community. 

This methodology follows Git Flow more than the Github Flow. See [https://lucamezzalira.com/2014/03/10/git-flow-vs-github-flow/](https://lucamezzalira.com/2014/03/10/git-flow-vs-github-flow/)

**Q: How long should a pending set of changes be under review?**
schema.org steering committee allows 10 days, but that seems short for our community. Maybe 30 days?

### Hotfixes & Bugs ###

schema.org steering group has the capability to bypass release candidates for what they call "Early Access Fixes". This seems similar to Git Flow's use of the the [hotfix](https://nvie.com/posts/a-successful-git-branching-model/#hotfix-branches) branch. 

### 4. Versioning Framework ###

We could adopt [Semantic Versioning](https://semver.org/)

Given a version number MAJOR.MINOR.PATCH, increment the:
* MAJOR version when you make incompatible API changes,
* MINOR version when you add functionality in a backwards-compatible manner, and
* PATCH version when you make backwards-compatible bug fixes.
Additional labels for pre-release and build metadata are available as extensions to the MAJOR.MINOR.PATCH format.

If the latest production tag was 1.2.5, the next release candidate would be 1.3.0-rc. Modifications to the release candidate would be 1.3.x-rc where x is incremented per modification. When the release candidate is ready for production the version tag becomes 1.3.0. 

* Hotfixes & Bugs (or Early Acceess Fixes) are always pushed as a PATCH to avoid collision with release candidates.

**Comment by rduerr** [comment]
Hmmm... what I don't really see here is the ESIP endorsement process which is, I think, an important part of Lewis's original proposal.  

What I am thinking is that a human readable (i.e., geared for repository managers and not programmers) summary of changes in a major release needs to be written and then submitted to the general assembly along with the release candidate.  If approved, we are talking DOI'ing the results and depositing in the ESIP repository (currently Figshare).

**Comment by ashepherd** [comment]
oh, thanks @rduerr ! I completely omitted the ESIP endorsement part. Do you have a vision of how that might work? I'm not aware of what process the Data Citation group went through for the ESIP blessing.

I like your idea about DOIs on the releases. We could use Zenodo to do that _from (Making your Code citable](https://guides.github.com/activities/citable-code/)_

**Comment by lewismc** [comment]
> I completely omitted the ESIP endorsement part. Do you have a vision of how that might work?

Having thought a bit more about this, I have envisioned a catch22. It comes down to the following;

1. do we do the work on solidifying earthsci.schema.org (BTW I am just using earthsci.schema.org to qualify alongside health-lifesci.schema.org we can change this anytime) first from the technical standpoint e.g. over at schema.org as a proposed extension OR, 
1. do we generate the human readable ESIP guideline version first based on [science-on-schema.org](https://github.com/ESIPFed/science-on-schema.org), publish it and then go to the schema.org community having done our homework, proving that we have a community and then propose it as an extension?

The reason I think this is important, is that ultimately our community and artifacts will reside in different places depending on which route we take. 

I personally would like to involve the [RDA data discovery community](https://github.com/RDA-DDP/Schema.org-for-Research-Data/blob/master/RDA-DDPIG-Schema-TF-Proposal.md) in all of this as well. My understanding here however is that their work has stalled. I do however think that this is because they did not get as far as @ashepherd and @fils. I think the RDA could play a significant and valuable role aiding in review of documentation here. 

So in summary the issues are
1. where are we aiming to have as a home? ESIP, schema.org or both?
1. Depending on what the answer is to the above, we need to create a strategy for getting there.

**Comment by lewismc** [comment]
> Q: Should we adopt a similar workflow? If so, how many people should sit on the committee?

I think this depends on where we aspire for this work to end up living. 
1. If it is over on schema.org, and if we aim to propose a hosted extension, then we will be fitting into their governance structure.
1. If it is ESIP, then I think we can bring together interested members and consider them as the committee. In all honesty, we tried to have a SWEET project management committee but it didn't really work. We worked around this by attempting to build as much consensus as we could on the Github issue tracker.

... or we could home it somewhere else which may result from RDA participation... who knows.

**Comment by lewismc** [comment]
> Q: How long should a pending set of changes be under review?
Yes I would suggest that we allow a telecon-to-telecon review period. We should also make review of pending earthsci.schema.org changes a recurring item on the telecon agenda.

**Comment by lewismc** [comment]
@ashepherd the above looks pretty solid to me. We can adapt it once we know whether we want  earthsci-schema.org to be a hosted or community extension. This will affect **Section 3 Workflow of Changes**.

**Comment by ashepherd** [comment]
I think I'm leaning towards doing what bioschemas.org has done. Going for an external extension, and then we have something that implementers can use right away. I think this will be quicker and help us gain some momentum. Then, if schema.org is interested in hosting all or part of that for us down the road, we can migrate over to geosci.schema.org.

**Comment by rduerr** [comment]
+1

**Comment by graybeal** [comment]
Definitely recommend external extension. In what I've heard about schema.org extensions, there is a strong bias in Google against hosting new extensions. And if Google hosts, there is inevitably much more process to deal with when making changes.

**Comment by charlesvardeman** [comment]
When @ashepherd and @lewismc had lunch at ESIP, I mentioned that something like a geoschemas.org was probably the best route vs possible issues with integrating into schema.org community governance. However, I recommend one possible side effect that we should be aware of is [standards creep](https://xkcd.com/927/) and maybe look at creating a set of best practice practices for linking to bioschemas.org and other science-related schema.org derivatives?

**Comment by ashepherd** [comment]
This issue has been obsoleted by issue #30 with the formation of the ESIP Schema.org cluster
